templates/common_macros.html.j2

```diff
@@ -859,3 +859,69 @@
     {% endif %}
   {% endfor %}
 {% endmacro %}
+
+{%- macro render_functional_chains(chains) -%}
+  {%- for chain in chains -%}
+    <h3>{{ linked_name_with_icon(chain) }}</h3>
+    {{ chain.context_diagram }}
+    <p>{{ description(chain) | safe }}</p>
+  {%- endfor -%}
+{%- endmacro -%}
+
+{%- macro get_cap_type(obj) -%}
+  {%- if obj.layer.xtype.endswith("PhysicalArchitecture") -%}
+    CapabilityRealization
+  {%- elif obj.layer.xtype.endswith("LogicalArchitecture") -%}
+    Capability
+  {%- else -%}
+    OperationalCapability
+  {%- endif -%}
+{%- endmacro %}
```

Comment:
This macro is only used to derive the `cap_type` argument for `render_functional_chain`. That however already receives the `object` in question anyway, so let's just inline this macro there (in the form of a few if/elif and set calls).